_examples/shell/help/README.md

```diff
@@ -0,0 +1,63 @@
+# Example of custom help providers
```

Comment:
I'd prefer this to just go in the main README, with illustrative examples in there.

Reply:
Thanks @alecthomas, yeah I figured that'd make sense - but didn't want to assume

If you're mostly happy with the copy and examples I can rework it to go in README - any recommendations on where?  Figure it could live directly under the existing section on help, or I could put it nearer the bottom and link from existing section.

One remaining query is around how/whether `HelpProvider` works with flags (and if so, how to show the additional help)

Reply:
Yeah under the existing help section would be idea, thanks.

HelpProvider does not currently work with flags, but I am midway through fixing that.

Reply:
Ping.

Reply:
Pong.

(Glad we're not playing quake)

Reply:
Finally moved examples into the main README

Have also removed the references to flags supporting `HelpProvider` for now to avoid confusion for others.

Started an exploratory branch to look at adding support for flags, but won't fit in my lunch break today!